frontend/src/queries/nodes/HogQLX/render.tsx

```diff
@@ -38,6 +43,30 @@ export function renderHogQLX(value: any): JSX.Element {
                     <Sparkline className="h-8" {...props} data={data ?? []} type={type} />
                 </ErrorBoundary>
             )
+        } else if (tag === 'RecordingButton') {
+            const { sessionId, ...props } = rest
+            const { openSessionPlayer } = useActions(sessionPlayerModalLogic)
+            return (
+                <ErrorBoundary>
+                    <LemonButton
+                        type="primary"
+                        size="xsmall"
+                        sideIcon={<IconPlayCircle />}
+                        data-attr="hog-ql-view-recording-button"
+                        {...props}
+                        to={urls.replaySingle(sessionId)}
+                        onClick={(e) => {
+                            e.preventDefault()
+                            if (sessionId) {
+                                openSessionPlayer({ id: sessionId })
+                            }
+                        }}
+                    >
+                        View recording
+                    </LemonButton>
+                    <SessionPlayerModal />
```

Comment:
Will this not add a modal for each rendered row? Maybe we could be clever and put this component in the header of the table to make sure it's rendered